pkg/controller/attach_detach.go

```diff
@@ -741,45 +729,47 @@ func (r *ReconcileAttachDetach) recoverAzVolumeAttachment(ctx context.Context, r
 		wg.Add(1)
 		go func(azv azdiskv1beta2.AzVolumeAttachment, azvMap *sync.Map) {
 			defer wg.Done()
+			var publishCtx map[string]string
+			if lun, _, err := r.cloudDiskAttacher.GetDiskLun(azv.Spec.VolumeName, azv.Spec.VolumeID, types.NodeName(azv.Spec.NodeName)); err == nil {
+				publishCtx = map[string]string{"LUN": strconv.Itoa(int(lun))}
+			}
 			var targetState azdiskv1beta2.AzVolumeAttachmentAttachmentState
-			updateMode := azureutils.UpdateCRIStatus
-			if azv.Spec.RequestedRole == azdiskv1beta2.ReplicaRole {
-				updateMode = azureutils.UpdateAll
+			if r.isAttached(ctx, azv.Spec.VolumeID, azv.Spec.NodeName) {
```

Comment:
Instead of checking if the disk is attached here, replace the `if (targetState == "")` statement on line 752 with an `if..else` statement, i.e.

```go
if (r.isAttached(...)) {
    targetState = azdiskv1beta2.Attached
} else {
    switch azv.Status.State {
        ...
    }
}
```

Splitting the initialization of `targetState` around the definition of the CRI update function makes it difficult to see the dependencies and could lead to errors in subsequent changes to this code.

Reply:
Done.